src/Chunk.cpp

```diff
@@ -1145,20 +1145,33 @@ int cChunk::GrowCactus(int a_RelX, int a_RelY, int a_RelZ, int a_NumBlocks)
 			for (auto & Coord : Coords)
 			{
 				if (
-					UnboundedRelGetBlockType(a_RelX + Coord.x, Top + 1, a_RelZ + Coord.z, BlockType) &&
+					UnboundedRelGetBlockType(a_RelX + Coord.x, Top + i, a_RelZ + Coord.z, BlockType) &&
 					(
 						cBlockInfo::IsSolid(BlockType) ||
 						(BlockType == E_BLOCK_LAVA) ||
 						(BlockType == E_BLOCK_STATIONARY_LAVA)
 					)
 				)
 				{
+					// Remove cactus
+					GetWorld()->DigBlock(a_RelX + GetPosX() * cChunkDef::Width, Top + i, a_RelZ + GetPosZ() * cChunkDef::Width);
+
+					// Drop cactus
+					cBlockHandler* Handler = BlockHandler(E_BLOCK_CACTUS);
+					cChunkInterface ChunkInterface(GetWorld()->GetChunkMap());
+					cBlockInServerPluginInterface PluginInterface(*GetWorld());
+					Handler->DropBlock(
+						ChunkInterface,
+						*GetWorld(),
+						PluginInterface,
+						nullptr,
+						a_RelX + GetPosX() * cChunkDef::Width  + (Coord.x * -1),
```

Comment:
??? `* -1` ???
Why not just `- Coord.x` ?

Reply:
I want to drop the block on the other side. On the left side is the stone on the right side is air, therefore *-1.

Reply:
That's worth a comment.

Anyway, I just wanted to point out that adding a number multiplied by -1 is the same as subtracting the number.